tests/benchmark_expr_templates.cpp

```diff
@@ -34,6 +34,29 @@ static std::pair<mfem::DenseMatrix, mfem::Vector> sample_matvec(const int rows,
   return {matrix, vec_in};
 }
 
+static auto build_partitioning(MPI_Comm comm, const int size)
+{
+  int num_procs = 0;
+  int rank      = 0;
+  MPI_Comm_size(comm, &num_procs);
+  MPI_Comm_rank(comm, &rank);
+  bool assumed_partition = HYPRE_AssumedPartitionCheck();
+  auto partitioning      = std::make_unique<int[]>(assumed_partition ? 2 : (num_procs + 1));
```

Comment:
hmm, I'm not actually sure what `std::make_unique<int[]>(assumed_partition ? 2 : (num_procs + 1));` does. Does this need `std::unique_ptr`, or does `std::vector<int>` work?

Reply:
The conditional size check is because Hypre uses one of two schemes for partitioning, depending on how it was compiled.

The constructor for `HypreParVector` only needs a raw integer array.  `std::vector` + `v.data()` would work fine, but `unique_ptr` is slightly less overhead.